src/Microsoft.AspNetCore.Routing/LinkGeneratorRouteValuesAddressExtensions.cs

```diff
@@ -0,0 +1,95 @@
+    {
+        public static string GetPathByRouteValues(
+            this LinkGenerator generator,
+            HttpContext httpContext,
+            string routeName,
+            object values,
+            FragmentString fragment = default,
+            LinkOptions options = default)
+        {
+            if (generator == null)
+            {
+                throw new ArgumentNullException(nameof(generator));
+            }
+
+            var address = CreateAddress(httpContext, routeName, values);
+            return generator.GetPathByAddress<RouteValuesAddress>(httpContext, address, address.ExplicitValues, fragment, options);
+        }
+
+        public static string GetPathByRouteValues(
+            this LinkGenerator generator,
+            string routeName,
+            object values,
+            PathString pathBase = default,
+            FragmentString fragment = default,
+            LinkOptions options = default)
+        {
+            if (generator == null)
+            {
+                throw new ArgumentNullException(nameof(generator));
+            }
+
+            var address = CreateAddress(httpContext: null, routeName, values);
+            return generator.GetPathByAddress<RouteValuesAddress>(address, address.ExplicitValues, pathBase, fragment, options);
+        }
+
+        public static string GetUriByRouteValues(
+            this LinkGenerator generator,
+            HttpContext httpContext,
+            string routeName,
+            object values,
+            FragmentString fragment = default,
+            LinkOptions options = default)
+        {
+            if (generator == null)
+            {
+                throw new ArgumentNullException(nameof(generator));
+            }
+
+            var address = CreateAddress(httpContext: null, routeName, values);
+            return generator.GetUriByAddress<RouteValuesAddress>(httpContext, address, address.ExplicitValues, fragment, options);
+        }
+
+        public static string GetUriByRouteValues(
+            this LinkGenerator generator,
+            string routeName,
+            object values,
+            string scheme,
+            HostString host,
+            PathString pathBase = default,
+            FragmentString fragment = default,
+            LinkOptions options = default)
+        {
+            if (generator == null)
+            {
+                throw new ArgumentNullException(nameof(generator));
+            }
+
+            var address = CreateAddress(httpContext: null, routeName, values);
+            return generator.GetUriByAddress<RouteValuesAddress>(address, address.ExplicitValues, scheme, host, pathBase, fragment, options);
+        }
+
+
+        private static RouteValuesAddress CreateAddress(HttpContext httpContext, string routeName, object values)
+        {
+            return new RouteValuesAddress()
+            {
+                AmbientValues = httpContext?.GetRouteData()?.Values,
```

Comment:
I've been wondering about `GetRouteData` and whether it should be considered legacy. I needed exactly `RouteData` to pass to MVC in `MvcEndpointDataSource` so I thought GetRouteData was acceptable but for getting just values I think we should use IRouteValuesFeature.